src/utils/url.ts

```
@@ -134,19 +169,25 @@ function delay(ms: number): Promise<void> {
 
 // Function to handle exponential backoff retries
 export async function retryWithExponentialBackoff<T>(
-  fn: () => Promise<T>,
-  resource: string,
-  retries = 5,
+  s3Fn: (config: S3Config) => Promise<T> | void,
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)